avm/src/vm/opcode.rs

```diff
@@ -3031,779 +3053,795 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       }
     };
     // Get the status from the user response and begin building the response object
-    let status = response_hm.read_fixed(0) as u16;
-    let mut res = Response::builder().status(StatusCode::from_u16(status).unwrap());
+    let status = response_hm.read_fixed(0)? as u16;
+    let mut res = Response::builder()
+      .status(StatusCode::from_u16(status).unwrap_or(StatusCode::INTERNAL_SERVER_ERROR));
     // Get the headers and populate the response object
+    // TODO: figure out how to handle this potential panic
     let headers = res.headers_mut().unwrap();
-    let header_hms = response_hm.read_fractal(1);
+    let header_hms = response_hm.read_fractal(1)?;
     for i in 0..header_hms.len() {
       let (h, _) = response_hm.read_from_fractal(&header_hms.clone(), i);
       let (key_hm, _) = response_hm.read_from_fractal(&h, 0);
       let (val_hm, _) = response_hm.read_from_fractal(&h, 1);
-      let key = HandlerMemory::fractal_to_string(key_hm);
-      let val = HandlerMemory::fractal_to_string(val_hm);
+      let key = HandlerMemory::fractal_to_string(key_hm)?;
+      let val = HandlerMemory::fractal_to_string(val_hm)?;
+      // TODO: figure out how to handle this potential panic
       let name = HeaderName::from_bytes(key.as_bytes()).unwrap();
+      // TODO: figure out how to handle this potential panic
       let value = HeaderValue::from_str(&val).unwrap();
       headers.insert(name, value);
     }
     // Get the body, populate the response object, and fire it out
-    let body = HandlerMemory::fractal_to_string(response_hm.read_fractal(2));
+    let body = HandlerMemory::fractal_to_string(response_hm.read_fractal(2)?)?;
+    // TODO: figure out how to handle this potential panic
     Ok(res.body(body.into()).unwrap())
   }
   io!(httplsn => fn(_args, hand_mem) {
     Box::pin(async move {
-      make_server!(&Program::global().http_config, http_listener);
-      return hand_mem;
+      // this extra fn is so that we can just use `?` inside of http_listener instead of
+      // having a bunch of `match`es that call a closure
+      async fn listen(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+        match http_listener(req).await {
+          Ok(res) => Ok(res),
+          Err(_) => {
+            // TODO: log the error?
```

Comment:
the chances to go through this brach is quite small right? If would represent an important error and difficult to track might be interesting to log it

Reply:
yes, but I'll let the daemon error-catching PR handle that